docs/quickstarts/ratify-on-azure.md

```diff
@@ -3,69 +3,241 @@ title: Ratify on Azure
+* Install [Cosign](https://docs.sigstore.dev/system_config/installation/) for signing container images with Cosign signatures
+* Install [Notation and Notation AKV plugin](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-tutorial-sign-build-push#install-notation-cli-and-akv-plugin) for signing container images with Notary Project signatures
+* Install and configure the latest [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli), or run commands in the [Azure Cloud Shell](https://portal.azure.com/#cloudshell/).
+
+### Configure environment variables
+
+```shell
+# Azure related variables
+export TENANT_ID=<your Tenant ID>
+export SUB_ID=<your subscription id>
+# AKV related variables
+export AKV_RG=<your AKV resource group>
+export AKV_NAME=<your AKV name>
+# The key used for Cosign signing
+export KEY_NAME=<name of the key to be created for Cosign signing>
+# ACR related variables
+export ACR_RG=<your ACR resource group>
+export ACR_NAME=<your ACR name>
+export IMAGE_SIGNED="$ACR_NAME.azurecr.io/ratify-demo/net-monitor:v1"
+export IMAGE_SIGNED_SOURCE=https://github.com/wabbit-networks/net-monitor.git#main
+export IMAGE_UNSIGNED="$ACR_NAME.azurecr.io/ratify-demo/net-watcher:v1"
+export IMAGE_UNSIGNED_SOURCE=https://github.com/wabbit-networks/net-watcher.git#main
+# Workload identity used by Ratify to access ACR and AKV
+export IDENTITY_NAME=<name of the identity to be created>
+export IDENTITY_RG=<your identity resource group name>
+# AKS related variables
+export AKS_RG=<your AKS resource group>
+export AKS_NAME=<your AKS name>
+export RATIFY_NAMESPACE="gatekeeper-system"
+```
 
-![e2e workflow diagram](../imgs/e2e-workflow-diagram.png)
+## Prepare container images in ACR
+
+> You can skip this section if you already built and pushed images to ACR.
+
+1. Sign in with Azure CLI
+
+    ```shell
+    az login
+    ```
+
+   To learn more about Azure CLI and how to sign in with it, see [Sign in with Azure CLI](https://learn.microsoft.com/en-us/cli/azure/authenticate-azure-cli).
+
+1. Ensure the logged-in identity has both `acrpush` and `acrpull` roles assigned.
+
+    ```shell
+    export USER_OBJECT_ID="$(az ad signed-in-user show --query id -o tsv)"
+    az role assignment create \
+        --assignee-object-id ${USER_OBJECT_ID} \
+        --role acrpull --role acrpush \
+        --scope subscriptions/${SUB_ID}/resourceGroups/${ACR_RG}/providers/Microsoft.ContainerRegistry/registries/${ACR_NAME}
+    ```
+
+1. Log into ACR
+
+    ```shell
+    az acr login --name ${ACR_NAME}
+    ```
+
+1. Build and push an image that will be signed in later steps.
+
+    ```shell
+    az acr build -r ${ACR_NAME} -t ${IMAGE_SIGNED} ${IMAGE_SIGNED_SOURCE} --no-logs
+    ```
+
+1. Build and push an image that will not be signed
+
+    ```shell
+    az acr build -r ${ACR_NAME} -t ${IMAGE_UNSIGNED} ${IMAGE_UNSIGNED_SOURCE} --no-logs
+    ```
+
+## Sign container images in ACR
+
```

Comment:
suggestion to let customer know they can choose one of notation or cosign signature to get started.